File: src/main/java/io/srmppn/HappyDorm/billing/api/PaymentVerification.java
package io.srmppn.HappyDorm.billing.api;

import org.axonframework.modelling.command.TargetAggregateIdentifier;

import java.util.UUID;

public class PaymentVerification {
    public static class VerifyPaymentCommand {
        @TargetAggregateIdentifier
        public UUID billId;
        public String paymentSecret;

        public VerifyPaymentCommand(UUID billId, String paymentSecret) {
            this.billId = billId;
            this.paymentSecret = paymentSecret;
        }

        public UUID getBillId() {
            return billId;
        }

        public String getPaymentSecret() {
            return paymentSecret;
        }
    }

    public static class PaymentVerifiedEvent {
        public UUID billId;
        public UUID roomId;

        public PaymentVerifiedEvent(UUID billId, UUID roomId) {
            this.billId = billId;
            this.roomId = roomId;
        }

        public UUID getBillId() {
            return billId;
        }

        public void setBillId(UUID billId) {
            this.billId = billId;
        }

        public UUID getRoomId() {
            return roomId;
        }

        public void setRoomId(UUID roomId) {
            this.roomId = roomId;
        }
    }
}

File: src/main/java/io/srmppn/happydorm/billing/api/payment-verification.kt
package io.srmppn.happydorm.billing.api

import org.axonframework.modelling.command.TargetAggregateIdentifier
import org.axonframework.serialization.Revision
import java.util.*


data class VerifyPaymentCommand(
  @TargetAggregateIdentifier
  val billId: UUID,
  val paymentSecret: String
)

@Revision("1.0")
data class PaymentVerifiedEvent(
  val billId: UUID,
  val roomId: UUID
)